pkg/injector/bootstrap_test.go

```diff
@@ -312,3 +325,233 @@ var _ = Describe("Test functions creating Envoy bootstrap configuration", func()
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notBootstrapSecret",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+			expSecrets: []*corev1.Secret{},
+		},
+	}
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Running test case %d: %s", i, tc.name), func(t *testing.T) {
+			assert := tassert.New(t)
+
+			mockController := k8s.NewMockController(gomock.NewController(t))
+			mockController.EXPECT().ListSecrets().Return(tc.secrets)
+
+			certManager, err := certificate.FakeCertManager()
+			assert.Nil(err)
+
+			b := NewBootstrapSecretRotator(mockController, certManager, time.Duration(1))
+
+			actual := b.listBootstrapSecrets()
+			assert.ElementsMatch(tc.expSecrets, actual)
+		})
+	}
+}
+
+func TestRotateBootstrapSecrets(t *testing.T) {
+	assert := tassert.New(t)
+
+	testNs := "testNamespace"
+	proxyUUID1 := uuid.New()
+	proxyUUID2 := uuid.New()
+	secretName := bootstrapSecretPrefix + proxyUUID1.String()
+	diffSecretName := bootstrapSecretPrefix + proxyUUID2.String()
+
+	testCases := []struct {
+		name      string
+		certNames []string
+		secrets   []*corev1.Secret
+	}{
+		{
+			name:      "update bootstrap secret with new cert",
+			certNames: []string{certificate.CommonName(proxyUUID1.String() + ".test.cert").String()},
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: testNs,
+					},
+					Data: map[string][]byte{
+						bootstrap.EnvoyXDSCACertFile: {},
+						bootstrap.EnvoyXDSCertFile:   []byte(certificates.SampleCertificatePEM),
+						bootstrap.EnvoyXDSKeyFile:    []byte(certificates.SamplePrivateKeyPEM),
+					},
+				},
+			},
+		},
+		{
+			name:      "update multiple bootstrap secret",
+			certNames: []string{certificate.CommonName(proxyUUID1.String() + ".test.cert").String(), certificate.CommonName(proxyUUID2.String() + ".test.cert").String()},
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: testNs,
+					},
+					Data: map[string][]byte{
+						bootstrap.EnvoyXDSCACertFile: {},
+						bootstrap.EnvoyXDSCertFile:   []byte(certificates.SampleCertificatePEM),
+						bootstrap.EnvoyXDSKeyFile:    []byte(certificates.SamplePrivateKeyPEM),
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      diffSecretName,
+						Namespace: testNs,
+					},
+					Data: map[string][]byte{
+						bootstrap.EnvoyXDSCACertFile: {},
+						bootstrap.EnvoyXDSCertFile:   []byte(certificates.SampleCertificatePEM),
+						bootstrap.EnvoyXDSKeyFile:    []byte(certificates.SamplePrivateKeyPEM),
+					},
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Running test case %d: %s", i, tc.name), func(t *testing.T) {
+			certManager, err := certificate.FakeCertManager()
+			assert.Nil(err)
+
+			mockController := k8s.NewMockController(gomock.NewController(t))
```

Comment:
Is it possible to create a fake `NewSimpleClientSet{[]runtime.Object{...}...)` for the kubeClient instead of mocking the controller? This would allow us to verify the secrets were updated as expected.

Reply:
+1 a SimpleClientset is the more robust approach